project/SparkBuild.scala

```diff
@@ -655,19 +655,47 @@ object SparkConnect {
 
     (assembly / logLevel) := Level.Info,
 
+    // Exclude `scala-library` from assembly.
+    (assembly / assemblyPackageScala / assembleArtifact) := false,
+
+    // Exclude `pmml-model-*.jar`, `scala-collection-compat_*.jar`,`jsr305-*.jar` and
+    // `unused-1.0.0.jar` from assembly.
+    (assembly / assemblyExcludedJars) := {
+      val cp = (assembly / fullClasspath).value
+      cp filter { v =>
+        val name = v.data.getName
+        name.startsWith("pmml-model-") || name.startsWith("scala-collection-compat_") ||
```

Comment:
Is `scala-collection` picked up from Scala library itself? I remember there's an option to exclude this (e.g., `includeScala = False`)

Reply:
should be `(assembly / assemblyPackageScala / assembleArtifact) := false`,  but I found `scala-collection` was not excluded, let me try

```
assembly / assemblyOption ~= {
      _.withIncludeScala(false)
    },
```

Reply:
<img width="870" alt="image" src="https://user-images.githubusercontent.com/1475305/194803168-f5f322ac-1226-4825-8642-721a4e1abd00.png">

@HyukjinKwon The results are the same, from the document, it should be this option, but it only excludes the `scala-library`, not the 'scala-collection compat_'.